File: trabalhos/T1/app/src/main/java/com/example/moviedata/data/MovieDao.kt
package com.example.moviedata.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface MovieDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insert(movie: Movie)

    @Query("SELECT * FROM movies WHERE title = :title")
    suspend fun getMovieByTitle(title: String): Movie?

    @Query("SELECT * FROM movies WHERE listType = :listType")
    suspend fun getMoviesByListType(listType: String): List<Movie>

    @Query("SELECT COUNT(*) FROM movies WHERE listType = :listType")
    suspend fun countMoviesInList(listType: String): Int

    @Query("SELECT * FROM movies WHERE title = :title AND listType = :listType")
    suspend fun getMovieByTitleAndListType(title: String, listType: String): Movie?

    @Query("DELETE FROM movies WHERE title = :title AND listType = :listType")
    suspend fun deleteMovieByTitleAndListType(title: String, listType: String)
}

File: trabalhos/T2/flutter1/lib/data/movie_dao.dart
import 'package:floor/floor.dart';
import 'movie.dart';

@dao
abstract class MovieDAO {
  @Insert(onConflict: OnConflictStrategy.replace)
  Future<void> insertMovie(Movie movie);

  @Query('SELECT * FROM movies WHERE title = :title')
  Future<Movie?> getMovieByTitle(String title);

  @Query('SELECT * FROM movies WHERE listType = :listType')
  Future<List<Movie>> getMoviesByListType(String listType);

  @Query('SELECT COUNT(*) FROM movies WHERE listType = :listType')
  Future<int?> countMoviesInList(String listType);

  @Query('SELECT * FROM movies WHERE title = :title AND listType = :listType')
  Future<Movie?> getMovieByTitleAndListType(String title, String listType);

  @Query('DELETE FROM movies WHERE title = :title AND listType = :listType')
  Future<void> deleteMovieByTitleAndListType(String title, String listType);

  @Query('UPDATE Movie SET listType = :listType WHERE title = :title')
  Future<void> updateMovieListType(String title, String listType);
}
